farm/eval.py

```diff
@@ -7,14 +7,23 @@
 from sklearn.metrics import classification_report
 from torch.utils.data import DataLoader
 
-from farm.metrics import compute_metrics
+from farm.metrics import compute_metrics, _correct_bio_encodings
 from farm.utils import to_numpy
 from farm.utils import MLFlowLogger as MlLogger
 from farm.modeling.adaptive_model import AdaptiveModel
 
 logger = logging.getLogger(__name__)
 
 
+def _token_classification_report(labels, preds, digits):
+    preds = [
+        [token if token not in {"X", "[PAD]"} else "O" for token in sent]
+        for sent in preds
+    ]
+    preds = _correct_bio_encodings(preds)
```

Comment:
I would either move this function in token_classification_report() or rename _token_classification_report() to corrected_token_classification_report() to not confuse people.